Mach-O test flakes due to "unexpected argument 'build'"

The following failure has happened twice today. The failed job succeeded when rerun. Not sure what's going on.

```
Run cargo build --profile ci --workspace --no-default-features
error: error: unexpected argument 'build' found

Usage: rustup-init[EXE] [OPTIONS]

For more information, try '--help'.

Error: Process completed with exit code 1.
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -167,6 +167,8 @@ jobs:
       - uses: dtolnay/rust-toolchain@stable
         id: rust-toolchain
       - uses: Swatinem/rust-cache@v2
+        with:
+          env-vars: ImageVersion
       - run: cargo build --profile ci --workspace --no-default-features
       - run: cargo test --profile ci --workspace --features macho
 

